diff --git a/mindspeed_mm/fsdp/train/train_engine.py b/mindspeed_mm/fsdp/train/train_engine.py
index a7f79fb9..73ce9c01 100644
--- a/mindspeed_mm/fsdp/train/train_engine.py
+++ b/mindspeed_mm/fsdp/train/train_engine.py
@@ -315,7 +315,7 @@ class TrainEngine:
         self.profiler.stop()
         memory_profiler.stop()
         # Final save after training completes
-        if args.training.save:
+        if args.training.save and args.training.save_interval > 0:   #加上了and后面的东西
             self.save(self.iteration, self.consumed_train_samples)
 
     def _run_validation(self, iteration):
